packages/hash/shared/src/blocks.ts

```diff
@@ -217,26 +214,47 @@ export const fetchBlockMeta = async (
       );
     }
 
-    const result: BlockMeta = {
-      componentMetadata: transformBlockConfig({
+    const result: HashBlock = {
+      meta: transformBlockConfig({
         metadata,
         schema,
         componentId: baseUrl,
       }),
-      componentSchema: schema,
+      schema,
     };
 
     return result;
   })();
 
-  if (typeof window !== "undefined") {
-    blockCache.set(baseUrl, promise);
-  }
+  prepareBlockCache(baseUrl, promise);
 
   return await promise;
 };
 
-export const blockComponentRequiresText = (
-  componentSchema: BlockMeta["componentSchema"],
-) =>
-  !!componentSchema.properties && "editableRef" in componentSchema.properties;
+const textBlocks = [
+  "https://blockprotocol.org/blocks/@hash/paragraph",
+  "https://blockprotocol.org/blocks/@hash/header",
+  "https://blockprotocol.org/blocks/@hash/callout",
+];
+
+/**
+ * Default blocks loaded for every user.
+ *
+ * @todo allow users to configure their own default block list, and store in db.
+ *    this should be a list of additions and removals from this default list,
+ *    to allow us to add new default blocks that show up for all users.
+ *    we currently store this in localStorage - see UserBlockProvider.
+ */
+export const defaultBlocks = [
+  ...textBlocks,
+  "https://blockprotocol.org/blocks/@hash/person",
+  "https://blockprotocol.org/blocks/@hash/image",
+  "https://blockprotocol.org/blocks/@hash/table",
+  "https://blockprotocol.org/blocks/@hash/divider",
+  "https://blockprotocol.org/blocks/@hash/embed",
+  "https://blockprotocol.org/blocks/@hash/code",
+  "https://blockprotocol.org/blocks/@hash/video",
+];
+
+export const isTextBlock = (componentId: string = "") =>
```

Comment:
I think we should add a comment saying this only checks against a set of defined blocks. This will return false for any text block that isn't in `textBlocks`

Reply:
I've added some comments, but also done on better. I realised in all but one place, this was only being used to check block compatibility, so I've built a better abstraction which is more explicit about this. I've also removed the other use because I realised it wasn't necessary. So we've ended up in what should be a much more readable position. It also fixes a bug where block switching is unexpectedly enabled when there's an error inside the block. [`a721d7a` (#872)](https://github.com/hashintel/hash/pull/872/commits/a721d7a92d7762568bb05b01f644d7b3387bc39a)